### PR TITLE
Adds ability to change encryption password of wallet.json

### LINKED ISSUE #651: Add ability to change encryption password of wallet.json
The encryption password can't be changed.
It wouldn't be too hard to add an option to wallet-tool.py which lets it be changed.

### PR DESCRIPTION
**Pull request by juscamarena**
This resolves #651
Suggestions welcome.
TODO: Add tests

### CODE REVIEW

File: wallet-tool.py [hunk 1]
```diff
@@ -229,6 +230,35 @@ def cus_print(s):
...
+    password_key = btc.bin_dbl_sha256(password)
+    encrypted_seed = encryptData(password_key, seed.decode('hex'))
+    timestamp = datetime.datetime.now().strftime("%Y/%m/%d %H:%M:%S")
+    walletname = args[0]
+    walletpath = os.path.join('wallets', walletname)
+    fd = open(walletpath, 'r')
+    walletfile = fd.read()
+    fd.close()
+    walletdata = json.loads(walletfile)
+    walletdata['creation_time'] = timestamp
```
Comment (AdamISZ) [inline wallet-tool.py]: Probably better not to change creation_time.

**Comment by AdamISZ** [comment]
tACK (just basic check).

I think this is fine (apart from the minor nit above). It would be better to factor out things like the password input and probably the wallet opening code, but no reason for that to block here (that kind of refactoring is needed in a few places).

Re: tests: the tests in test/test_wallets.py were written ages ago (I started out using class-based tests but that isn't necessary with the py.test approach we switched to, just functions is fine). Although the tests for wallets are quite crappy and neglected, the basic idea of using pexpect as [here](https://github.com/JoinMarket-Org/joinmarket/blob/master/test/test_wallets.py#L225-L231) should be fine for creating a test where you run the `changepassphrase` command, and then check that the wallet still opens. What's really needed there (and same for existing tests) is to have a sanity check of balance after applying the command. If I had time I would improve it myself.

File: wallet-tool.py (hunk 1 above)
Comment (chris-belcher) [inline wallet-tool.py]: Yes, that wallet creation time one day needs to be used for #470 and https://github.com/JoinMarket-Org/joinmarket/issues/55#issuecomment-184165679

Comment (juscamarena) [inline wallet-tool.py]: Ah, good to know! I can change this.

**Comment by juscamarena** [comment]
I wanted to touch as little of the rest of the code as I could for a first time pull request, but yeah I agree with  factoring most of the wallet related io out.

**Comment by chris-belcher** [comment]
Looks great to me.

Squish these two commits together (here's a tutorial in case it helps http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)

And also add tests. I can help you on IRC since I remember getting them to run is a little tricky. There's also this tutorial on the wiki https://github.com/JoinMarket-Org/joinmarket/wiki/Testing

**Comment by chris-belcher** [comment]
Shall we merge this? (Reminder to self that the merge.py script needs to be used so every commit is signed)

**Comment by juscamarena** [comment]
I had the test working and passing with the correct outputs being checked with logs.. When I run them locally now I'm getting weird issues with bitcoind in regtest requesting wallet history instead of executing my changepassphrase function.. I'm stumped.

**Comment by AdamISZ** [comment]
The test is expecting ": passphrase has been updated" but that is not printed on the console, and I can't find it in the code? Was it something you changed/removed perhaps?

**Comment by juscamarena** [comment]
Okay, now I think it's ready to merged.

**Pull request merged**